Wrappers/Python/ccpi/viewer/utils/conversion.py

```diff
@@ -753,6 +753,13 @@ def SetIsAcquisitionData(self, value):
     def GetIsAcquisitionData(self):
         return self.__IsAcquisitionData
 
+    def ReadDataSetInfo(self):
+        '''Not implemented for Base class
+        Tries to read info about dataset from given file
+        Will raise specific errors if inputs required for 
+        file type are not set.'''
+        pass
```

Comment:
`raise NotImplementedError` instead of `pass`

Reply:
The ReadDataSetInfo gets called in the base class because currently the base class is used for reading raw data
So instead of not implemented error, which would break the reading of raw data, I added a check to make sure the user set the size of the dataset and the typecode.
I think we should move the specifics of raw image reading out of the base class to a RawResampleReader, so that the Baseclass is a true baseclass for all readers that is never called itself. Shall I make a separate issue for this?

Reply:
Yes please